x/upgrade/abci_test.go

```diff
@@ -0,0 +1,171 @@
+package upgrade
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+)
+
+type TestSuite struct {
```

Comment:
I really like this testing format 👍 👍

Reply:
I wish all of our begin/end blockers were tested this well!